rpc/v8/transaction.go

```diff
@@ -364,29 +366,53 @@ func adaptBroadcastedTransaction(broadcastedTxn *BroadcastedTransaction,
 	return txn, declaredClass, paidFeeOnL1, nil
 }
 
-func adaptResourceBounds(rb map[core.Resource]core.ResourceBounds) map[Resource]ResourceBounds {
-	rpcResourceBounds := make(map[Resource]ResourceBounds)
-	for resource, bounds := range rb {
-		rpcResourceBounds[Resource(resource)] = ResourceBounds{
-			MaxAmount:       new(felt.Felt).SetUint64(bounds.MaxAmount),
-			MaxPricePerUnit: bounds.MaxPricePerUnit,
+func adaptResourceBounds(rb map[core.Resource]core.ResourceBounds) *TransactionResourceBounds {
+	if len(rb) == 0 {
+		return &TransactionResourceBounds{}
+	}
+
+	return &TransactionResourceBounds{
+		L1Gas:     getResourceBounds(rb, core.ResourceL1Gas),
+		L1DataGas: getResourceBounds(rb, core.ResourceL1DataGas),
+		L2Gas:     getResourceBounds(rb, core.ResourceL2Gas),
+	}
+}
+
+func getResourceBounds(rb map[core.Resource]core.ResourceBounds, resource core.Resource) *ResourceBounds {
+	if coreBounds, ok := rb[resource]; ok {
+		return &ResourceBounds{
+			MaxAmount:       new(felt.Felt).SetUint64(coreBounds.MaxAmount),
+			MaxPricePerUnit: coreBounds.MaxPricePerUnit,
 		}
 	}
-	return rpcResourceBounds
+	return nil
 }
 
-func adaptToFeederResourceBounds(rb *map[Resource]ResourceBounds) *map[starknet.Resource]starknet.ResourceBounds { //nolint:gocritic
+func adaptToFeederResourceBounds(rb *TransactionResourceBounds) *map[starknet.Resource]starknet.ResourceBounds { //nolint:gocritic
```

Comment:
maps are already reference types, it is no good return a reference to a reference. Specially if that map is being created inside this same function. Changing the function signature defintion to

```go
func adaptToFeederResourceBounds(rb *TransactionResourceBounds) map[starknet.Resource]starknet.ResourceBounds {
```

shouldn't incur in any extra changes in the function body